shell/common/fixtures/shell_test.dart

```diff
@@ -529,3 +531,45 @@ void testReportViewWidths() {
     nativeReportViewWidthsCallback(getCurrentViewWidths());
   };
 }
+
+@pragma('vm:entry-point')
+void renderWarmUpImplicitViewAfterMetricsChanged() {
+  PlatformDispatcher.instance.onBeginFrame = (Duration beginTime) {
+    // Make sure onBeginFrame and onDrawFrame are not used later.
+    PlatformDispatcher.instance.onBeginFrame = null;
+    PlatformDispatcher.instance.onDrawFrame = null;
+    // Let the test know that a continuation is available.
+    notifyNative();
+  };
+
+  bool beginFrameDone = false;
+
+  // As soon as 2 views are added, render these views.
+  PlatformDispatcher.instance.onMetricsChanged = () {
+    PlatformDispatcher.instance.scheduleWarmUpFrame(
```

Comment:
This seems like a strange test to me - this API is documented as saying not to call it at arbitrary times, and metrics changes can happen for all sorts of reasons.

If anything, I'd expect we make this test fail at some point rather than continue to support it, right?

Reply:
I've managed to simplify the test. Now the dart program simply schedules the warm up frame immediately, since I realized that the frame can be scheduled from the engine side to create the continuation.